File: frontend/types/team.ts
export type TeamStatus =
  | "Pending"
  | "Approved"
  | "Active"
  | "Eliminated"
  | "Withdrawn"
  | "Champion"
  | "Archived";

export interface TeamMember {
  userId: string;
  fullName: string;
  email: string;
  role?: "Member" | "Leader";
  isLeader?: boolean;
}

export interface TeamCategory {
  categoryId: string;
  categoryName: string;
}

export interface TeamRound {
  roundId: string;
  roundName: string;
}

export interface Team {
  teamId: string;
  teamName: string;
  description?: string | null;
  status: TeamStatus;
  leaderId?: string;
  category: TeamCategory;
  currentRound?: TeamRound | null;
  members: TeamMember[];
  createdAt?: string;
  eliminationReason?: string | null;
  statusReason?: string | null;
  eliminatedAt?: string | null;
}

export interface CreateTeamRequest {
  teamName: string;
  categoryId: string;
  memberIds: string[];
}

export interface EliminateTeamRequest {
  reason: string;
}

export interface AddTeamMemberRequest {
  userId: string;
}

export interface UpdateTeamRequest {
  teamName: string;
  description?: string;
  categoryId: string;
}

export interface InviteTeamMemberRequest {
  userId?: string;
  email?: string;
}

export interface TeamInvite {
  teamInviteId: string;
  createdAt: string;
  expiresAt: string;
  status: "Pending" | "Accepted" | "Rejected" | "Cancelled" | "Expired";
  team: {
    teamId: string;
    teamName: string;
    category: string;
  };
}

export interface TransferLeadershipRequest {
  newLeaderUserId: string;
}

File: Backend/Models/Entities/Team.cs
using SEAL.NET.Models.Enums;

namespace SEAL.NET.Models.Entities
{
    public class Team
    {
        public Guid TeamId { get; set; } = Guid.NewGuid();
        public string TeamName { get; set; } = string.Empty;
        public string? Description { get; set; }
        public TeamStatus Status { get; set; } = TeamStatus.Pending;
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public string? EliminationReason { get; set; }
        public string? StatusReason { get; set; }
        public DateTime? EliminatedAt { get; set; }
        public bool IsArchived { get; set; } = false;

        public Guid LeaderId { get; set; }
        public ApplicationUser Leader { get; set; } = null!;

        public Guid? CurrentRoundId { get; set; }
        public Round? CurrentRound { get; set; }

        public Guid CategoryId { get; set; }
        public Category Category { get; set; } = null!;

        public List<TeamMember> Members { get; set; } = [];
        public List<Submission> Submissions { get; set; } = [];
        public List<TeamInvite> Invites { get; set; } = [];
        public List<MentorAssignment> MentorAssignments { get; set; } = [];
        public List<MentorshipNote> MentorshipNotes { get; set; } = [];
    }
}
